typescript/api/test/unit/services/FigshareService.test.ts

```diff
@@ -0,0 +1,135 @@
+const { describe, it, beforeEach } = require('mocha');
+const { expect } = require('chai');
+
+// Minimal sails stub required by FigshareService
+(global as any).sails = {
+  config: {
+    figshareAPIEnv: {
+      overrideArtifacts: {
+        mapping: {
+          artifacts: {}
+        }
+      }
+    },
+    figshareAPI: {
+      APIToken: 'test-token',
+      baseURL: 'https://api.figshare.test',
```

Comment:
Are these real urls: 'https://api.figshare.test', 'https://figshare.example' ?
If they aren't, or aren't meant for testing, I suggest using urls we control as examples instead, just in case the requests accidentally get sent.